pkg/mons/mon_endpoints.go

```diff
@@ -27,11 +27,8 @@ import (
 
 const monConfigMap = "rook-ceph-mon-endpoints"
 
-func GetMonEndpoint(clusterNamespace string) {
-	// Get Kubernetes Client
-	_, _, client := k8sutil.GetKubeClient()
-
-	monCm, err := client.ConfigMaps(clusterNamespace).Get(context.TODO(), monConfigMap, v1.GetOptions{})
+func GetMonEndpoint(ctx *k8sutil.Context, clusterNamespace string) {
```

Comment:
The rook pattern is to import like this so the `context` var name can be used consistently:
```
ctx "context"
```

```suggestion
func GetMonEndpoint(context *k8sutil.Context, clusterNamespace string) {
```

Reply:
Is there a unit test that could use a fake k8s context now? In a separate PR is fine though.